### ISSUE TITLE
Per Head Max-Attention-Logits

### ISSUE THREAD

**Issue by ajWithNucleus** [Dao-AILab/flash-attention#1876]
Muon optimizers are train and scale well for transformers. Recent kimi K2 paper introduced Muon with QK clipping which helps in stabilizing large models. 

To implement this, we need per head max attention logits. What’s the reliable way to get this information without sacrificing performance.

**Comment by tridao** [comment]
At the end of the fwd pass we have the softmax row_max and row_sum. We then compute lse in `softmax.finalize`, then write that lse to global memory. You can additional write the `softmax.row_max` to global memory in the same way that we write LSE. Note that this is the row_max before multiplying by softmax_scale (typically 1/sqrt(d)). 
https://github.com/Dao-AILab/flash-attention/blob/dfb664994c1e5056961c90d5e4f70bf7acc8af10/hopper/flash_fwd_kernel_sm90.h#L444

**Comment by NJX-njx** [comment]
I'd like to work on this. Plan is to write \ to global memory alongside LSE in \, and expose it through a \ parameter in the interface. Will submit a PR.
